docs/pages/configuration.md

```diff
@@ -0,0 +1,611 @@
+    'store' => [
+       'type' => 'dbal_stream',
+    ],
+];
+```
+
+Following store types are available:
+
+- `dbal_aggregate` *default (deprecated)*
+- `dbal_stream` *recommended*
+- `in_memory`
+- `custom`
+
+!!! note
+
+    If you use `custom` store type, you need to set the service id under `store.service`.
+    
+### Change table Name
+
+You can change the table name of the event store.
+
+```php
+return [
+    'store' => [
+       'type' => 'dbal_stream',
+       'options' => [
+           'table_name' => 'my_event_store',
+       ],
+    ],
+];
+```
+
+### Read Only Mode
+
+For `dbal_aggregate` and `dbal_stream` store types you can activate the read only mode.
+Readings are possible, but if you try to write, an exception `StoreIsReadOnly` is thrown.
+
+```php
+return [
+    'store' => [
+       'type' => 'dbal_stream',
+       'readonly' => true,
+    ],
+];
+```
+
+!!! tip
+
+    This is useful if you have maintenance work on the event store and you want to avoid side effects.
+
+### Data Migration
+
+If you want to migrate from your current store to a new store, you can use the following configuration.
+This registers a new store and a new cli command `event-sourcing:store:migrate`.
+You can define translators to translate the old events to the new store.
+Here is an example for a migration from `dbal_aggregate` to `dbal_stream`.
+
+```php
+return [
+    'store' => [
+       'type' => 'dbal_aggregate',
+       'readonly' => true,
+       'options' => [
+           'table_name' => 'old_store',
+       ],
+       'migrate_to_new_store' => [
+           'enabled' => true,
+           'type' => 'dbal_stream',
+           'options' => [
+               'table_name' => 'my_stream_store',
+           ],
+           'translators' => [
+               Patchlevel\EventSourcing\Message\Translator\AggregateToStreamHeaderTranslator::class,
+           ],
+       ],
+    ],
+];
+```
+
+!!! danger
+
+    Make sure that you use different table names for the old and new store.
+    Otherwise your event store will be destroyed.
+    
+!!! tip
+
+    Set the `read_only` flag to `true` for the old store to avoid side effects
+    and missing events during the migration.
+
+## Subscription
+
+!!! tip
+
+    You can find out more about subscriptions in the library 
+    [documentation](https://event-sourcing.patchlevel.io/latest/subscription/).
+    
+### Store
+
+You can change where the subscription engine stores its necessary information about the subscription.
+Default is `dbal`, which means it stores it in the same DB that is used by the dbal event store.
+
+Otherwise you can choose between the following stores:
+
+- `dbal` *default*
+- `in_memory`
+- `static_in_memory`
+- `custom`
+
+
+```php
+return [
+    'subscription' => [
+       'store' => [
+           'type' => 'custom', # default is 'dbal'
+           'service' => 'my_subscription_store',
+       ],
+    ],
+];
+```
+
+// @TODO needs to make it compatible with `RefreshDatabase` i think
```

Comment:
Laravel provides a trait for transaction wrapping of test methods. I think we somehow need to register ourself into that, so the eventstore can be reset by using this trait, wdyt?

Reply:
Can't do that, due to not using an eloquent connection.

Need to think about if we can provide something to ease testing. But for now out of scope of this PR